bootstrap/config.go

```diff
@@ -59,7 +59,7 @@ type Config struct {
 	AgentName string
 
 	// Should the bootstrap remove an existing checkout before running the job
-	CleanCheckout bool
+	CleanCheckout bool `env:"BUILDKITE_CLEAN_CHECKOUT"`
```

Comment:
I initially didn't understand how adding this made sense.

The hooks changing `BUILDKITE_CLEAN_CHECKOUT` were being run _inside_ the bootstrap process, not prior to it running, so how does passing it to the bootstrap as environment help?

Then I realised this gets called after each hook: https://github.com/buildkite/agent/blob/75c55cb5c99d8635d69ed6456b6643de861a0a95/bootstrap/bootstrap.go#L253

So we're not really expecting it to be passed to the bootstrap as environment, but we do want the internal config `ReadFromEnvironment()` to understand it in the fake environment that we build after running hooks.

👌🏼